server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -573,16 +574,18 @@ export class AssetService {
   private getThumbnailPath(asset: AssetEntity, format: GetAssetThumbnailFormatEnum) {
     switch (format) {
       case GetAssetThumbnailFormatEnum.WEBP:
-        if (asset.webpPath && asset.webpPath.length > 0) {
-          return asset.webpPath;
+        if (!asset.webpPath) {
+          throw new NotFoundException(`No ${format} thumbnail found for asset ${asset.id}`);
         }
+        return asset.webpPath;
 
       case GetAssetThumbnailFormatEnum.JPEG:
-      default:
         if (!asset.resizePath) {
-          throw new NotFoundException('resizePath not set');
+          throw new NotFoundException(`No ${format} thumbnail found for asset ${asset.id}`);
         }
         return asset.resizePath;
+      default:
```

Comment:
Does anything rely on this behaviour?

Reply:
From what I can tell, the main instance where the fallback would come up is in the narrow window where the jpeg thumbnail is generated but the webp thumbnail is still queued.

Reply:
For a large list of assets jpeg could be generated and then webp gets added to the end of the queue and might take awhile before it is processed. If you have 10k photos queued the jpeg is generated and the webp becomes task 10,001. The mobile app gets a notification after the jpeg is generated so it might try to access some form of thumbnail after that. Not sure if it asks for webp or not. Would be worth checking.

Reply:
Every usage of this endpoint queries it with `format` omitted (defaulting to webp). Would it make sense to switch the order of those jobs in that case so webp is generated first?

Reply:
Webp is generated from the jpeg right now...

Reply:
This is just a rabbit hole imo requiring bigger changes which we could avoid by _only_ fixing content type header without changing any other behavior.

Reply:
Fair enough, I'll have it validate that it's a valid enum value but keep the fallback to jpeg.